### PR TITLE
Add values for dynamic backend plugin support

### LINKED ISSUE #118: Integrate Dynamic Plugins support in helm charts

### PR DESCRIPTION
**Pull request by davidfestal**
Signed-off-by: David Festal <dfestal@redhat.com>

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [pre-commit](https://pre-commit.com/) utility can be used to generate the necessary content. Use `pre-commit run -a` to apply changes.
- [x] JSON Schema template updated and re-generated the raw schema via `pre-commit` hook.
- [x] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

### CODE REVIEW

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:22]: Can you use `upstream.backstage.extraAppConfig` value instead?

https://github.com/backstage/charts/tree/main/charts/backstage#pass-extra-configuration-files

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:55]: No need for this if you use the suggestion made above.

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:64]: No need for this if you use the suggestion made above.

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:122]: Can we move this into the `charts/backstage/templates/` folder as a standalone template?

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:71]: As you probably know already, this will not work 🙂 (just stating the obvious, so we don't overlook it)

**Review by tumido — CHANGES_REQUESTED** [review]

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml:71]: Yeah, sure. That's part of why the PR is still WIP :-)
But that's good that you mentioned it, so that we don't forget changing it before merging !

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml:21]: I don't think it is possible, since this additional app-config file is generated by the installation script in the init container, based on the content of the `dynamic-plugins.yaml` file (provided through the `dynamic-plugins`  config map, but originally defined in the `global.dynamicPlugins` Helm value.

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml:59]: I don't see why. This config map doesn't contain a backstage `app-config` file, but the list of dynamic plugins used by the init container script to install the dynamic plugins and merge their `app-config` fragments to the additional `app-config.dynamic-plugins.yaml` file.

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml]: Done.

File: charts/backstage/values.yaml
Comment (Zaperex) [reply charts/backstage/values.yaml:23]: @davidfestal I don't think this method would work since these new arguments are not applied unless you also change the `command` line from `command: []` to `command: ['node', 'packages/backend']` to override the [entrypoint](https://github.com/janus-idp/backstage-showcase/blob/0d27cbbda96f6ae2bde21bb1540a3e326b84ac18/docker/Dockerfile#L138) defined when building the container image. 

However, if you do just add `command: ['node', 'packages/backend']`  alongside the edit you made above, it won't apply the `app-config.yaml` and other config files that are normally applied by default, which can cause backend to fail due to missing configurations.

File: charts/backstage/values.yaml
Comment (Zaperex) [inline charts/backstage/values.yaml:23]:
```suggestion
    command:
      - node
      - packages/backend
    args:
      - '--config'
      - 'app-config.yaml'
      - '--config'
      - 'app-config.example.yaml'
      - '--config'
      - 'app-config.example.production.yaml'
      - '--config'
      - dynamic-plugins-root/app-config.dynamic-plugins.yaml
```
As mentioned above, it would need to be something like this

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:19]: please default to `latest`

File: charts/backstage/values.yaml
Comment (tumido) [reply charts/backstage/values.yaml:23]: I don't think we need to override the `command`. Kubernetes starts the pod via `[ENTRYPOINT] [CMD]` which translates to `[command] [args]`, so this solution should work fine.

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:51]: Can you please explain/document/comment on these inline?

File: charts/backstage/values.yaml
Comment (tumido) [inline charts/backstage/values.yaml:3]: Please comment on where is this file located (basepath for `includes`) and that it only works with backstage-showcase image.

**Review by tumido — CHANGES_REQUESTED** [review]
I have only minor cosmetic issues, see comments inline.

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml:23]: Ad @tumido mentioned in [this comment](https://github.com/janus-idp/helm-backstage/pull/111/files#r1370738865) it doesn't seem necessary.

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml]: Done

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml:61]: Done

File: charts/backstage/values.yaml
Comment (davidfestal) [reply charts/backstage/values.yaml]: Done, but afaict that will the CI tests. Should I temporarily change the tags in [CI default values](git push origin dynamic-backend-plugins:dynamic-backend-plugins) ?

Comment (davidfestal) [reply charts/backstage/values.yaml]: In fact I have temporarily changed the CI test `initContainer` image tag to `next` in commit c20c21b66d1df465ba2a079834f562f16aa2f82d.
With that tests should succeed.

**Comment by Zaperex** [comment]
Will also need to add the `integrity` field into the helm chart schema after https://github.com/janus-idp/backstage-showcase/pull/670 is merged

**Comment by davidfestal** [comment]
@tumido Is the PR OK for you now ?

I assume we'll have to wait for the new release of the showcase before merging it ?

File: .github/workflows/test.yml
Comment (tumido) [inline .github/workflows/test.yml:71]: Please revert this to `latest`

**Review by tumido — CHANGES_REQUESTED** [review]
One last revert required, otherwise lgtm.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Comment by Zaperex** [comment]
@davidfestal we still need to add the integrity field into the helm chart schema since https://github.com/janus-idp/backstage-showcase/pull/670 was merged.

**Pull request merged**
